### DISCUSSION TITLE
Pimcore 11: Grid view search displaying empty containers instead of relevant result

### DISCUSSION THREAD

**Discussion by sandyHM** [pimcore/pimcore#17120] [Admin-Ui-Classic-Bundle]
Grid view search displaying empty containers instead of relevant result.

![screenshot-pim11 contitech local-2024 06 04-09_42_34](https://github.com/pimcore/pimcore/assets/107986459/21d0c1da-0f7f-4260-9d9c-5a817087e7ab)

**Comment by sandyHM** [comment]
@kingjia90 ??

**Reply by sandyHM** [reply]
@kingjia90 ??

**Reply by kingjia90** [reply]
Converting to an issue https://github.com/pimcore/admin-ui-classic-bundle/issues/637 but doesn't look that critical
Could you please provide a PR for it ?

Somewhere here is not right
https://github.com/pimcore/admin-ui-classic-bundle/blob/f7a8003729ff83244bc005b8afa5c2206c695778/public/js/pimcore/object/helpers/classTree.js#L41-L61

It looks like the `Layout Components` are necessary to navigate, but could be hidden if they don't have any children
![image](https://github.com/user-attachments/assets/73250479-eab7-4c8b-be2d-0bc005cc0d98)

**Reply by sandyHM** [reply]
This should be hide in Grid View section, it's really very annoying and taking too much time to find any attribute in long list. Please look into it.

**Reply by kingjia90** [reply]
As it is only a visual annoyance, it is unlikely to be fixed anytime soon.
To speed up the fix, i would like to recommend you to provide a PR for it and i'd glad to help reviewing it. 

Thank you in advance

**Reply by sandyHM** [reply]
@kingjia90 
i added below code in updateFilter function in  public/js/pimcore/object/helpers/classTree.js file .
`rootNode.collapse(true);
            rootNode.expand(true);`
after applying mentioned changes, not getting any empty container.

**Reply by kingjia90** [reply]
Thank you for your followup on this, that's an interesting solution (if it works, it works 😄), could you please provide a PR so that the fix would be credited to you? TIA

**Reply by sandyHM** [reply]
Hi @kingjia90 
mentioned solution only working with search button not with onkeyup. I think this is not a proper solution.

**Reply by kingjia90** [reply]
ah, i see, please open a draft PR and continue the conversation on https://github.com/pimcore/admin-ui-classic-bundle/issues/637 as this discusssion post is closed
